go/vt/mysqlctl/mysqld.go

```diff
@@ -1195,9 +1214,18 @@ func (mysqld *Mysqld) GetVersionComment(ctx context.Context) string {
 	return versionComment
 }
 
-// applyBinlogFile extracts a binary log file and applies it to MySQL. It is the equivalent of:
+// ApplyBinlogFile extracts a binary log file and applies it to MySQL. It is the equivalent of:
 // $ mysqlbinlog --include-gtids binlog.file | mysql
-func (mysqld *Mysqld) applyBinlogFile(binlogFile string, includeGTIDs mysql.GTIDSet) error {
+func (mysqld *Mysqld) ApplyBinlogFile(ctx context.Context, binlogFile string, restorePos mysql.Position) error {
+	if socketFile != "" {
+		log.Infof("executing Mysqld.ApplyBinlogFile() remotely via mysqlctld server: %v", socketFile)
+		client, err := mysqlctlclient.New("unix", socketFile)
+		if err != nil {
+			return fmt.Errorf("can't dial mysqlctld: %v", err)
+		}
+		defer client.Close()
+		return client.ApplyBinlogFile(ctx, binlogFile, mysql.EncodePosition(restorePos))
+	}
```

Comment:
This is now part of the interface and also has a remote version so we can run this correctly for restores through `mysqlctl`.